packages/apollo-link-utilities/src/index.ts

```diff
@@ -0,0 +1,269 @@
+import { Operation } from 'apollo-link';
+import { print } from 'graphql/language/printer';
+
+// XXX replace with actual typings when available
+declare var AbortController: any;
+
+//Used for any Error for data from the server
+//on a request with a Status >= 300
+//response contains no data or errors
+export type ServerError = Error & {
+  response: Response;
+  result: Record<string, any>;
+  statusCode: number;
+};
+
+//Thrown when server's resonse is cannot be parsed
+export type ServerParseError = Error & {
+  response: Response;
+  statusCode: number;
+  bodyText: string;
+};
+
+export type ClientParseError = Error & {
+  parseError: Error;
+};
+
+export interface HttpOptions {
+  includeQuery?: boolean;
+  includeExtensions?: boolean;
+}
+
+export interface ConfigOptions {
+  http?: HttpOptions;
+  options?: any;
+  headers?: any; //overrides headers in options
+  credentials?: any;
+}
+
+export namespace LinkUtils {
+  export interface UriFunction {
+    (operation: Operation): string;
+  }
+
+  export interface Options {
+    /**
+     * The URI to use when fetching operations.
+     *
+     * Defaults to '/graphql'.
+     */
+    uri?: string | UriFunction;
+
+    /**
+     * Passes the extensions field to your graphql server.
+     *
+     * Defaults to false.
+     */
+    includeExtensions?: boolean;
+
+    /**
+     * A `fetch`-compatible API to use when making requests.
+     */
+    fetch?: GlobalFetch['fetch'];
+
+    /**
+     * An object representing values to be sent as headers on the request.
+     */
+    headers?: any;
+
+    /**
+     * The credentials policy you want to use for the fetch call.
+     */
+    credentials?: string;
+
+    /**
+     * Any overrides of the fetch options argument to pass to the fetch call.
+     */
+    fetchOptions?: any;
+  }
+
+  const defaultHttpOptions: HttpOptions = {
+    includeQuery: true,
+    includeExtensions: false,
+  };
+
+  const defaultHeaders = {
+    // headers are case insensitive (https://stackoverflow.com/a/5259004)
+    accept: '*/*',
```

Comment:
I know it has always been `*/*`, but shouldn't it really be `application/json`, since that is what is required to work?